dashboard/app/models/course_version.rb

```diff
@@ -151,6 +152,25 @@ def recommended?(locale_code = 'en-us')
     latest_stable_version == content_root
   end
 
+  def self.courses_for_unit_selector(unit_ids, user)
```

Comment:
Why does this method depend on user?  Do we want to hide courses in the section progress dashboard that the teacher does not have access to?  Would we consider showing them if it simplifies our logic here?

For the future, I think it would be helpful to pull the filtering by user "up" as far as possible so that as few methods as possible need the user as input.  (For this particular instance, another option would be to filter by unit_ids first and only pass the list of "acceptable" unit ids to these various functions.)

Reply:
I like the idea of filter by unit_ids first and only pass the list of "acceptable" unit ids to these various functions. I can take a look at that